zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -283,6 +342,154 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
+                                    confidenceId,
+                                    desc,
+                                    updatedAlert.getUri(),
+                                    param,
+                                    attack,
+                                    otherInfo,
+                                    solution,
+                                    refs,
+                                    evidence,
+                                    cweId,
+                                    wascId,
+                                    updatedAlert.getHistoryRef().getHistoryId());
+                } catch (DatabaseException e) {
+                    logger.error(e.getMessage(), e);
+                    throw new ApiException(ApiException.Type.INTERNAL_ERROR, e);
+                }
+            }
+        } else if (ACTION_ADD_ALERT.equals(name)) {
+            int messageId = ApiUtils.getIntParam(params, PARAM_ALERT_MESSAGE_ID);
+            String alertName = ApiUtils.getNonEmptyStringParam(params, PARAM_ALERT_NAME);
+            int riskId = ApiUtils.getIntParam(params, PARAM_ALERT_RISK);
+            int confidenceId = ApiUtils.getIntParam(params, PARAM_ALERT_CONFIDENCE);
+            String desc = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_DESCRIPTION);
+            String param = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_PARAM);
+            String attack = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_ATTACK);
+            String otherInfo = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_OTHERINFO);
+            String solution = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_SOLUTION);
+            String refs = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_REFS);
+            String evidence = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_EVIDENCE);
+            int cweId = ApiUtils.getIntParam(params, PARAM_ALERT_CWEID);
+            int wascId = ApiUtils.getIntParam(params, PARAM_ALERT_WASCID);
+
+            TableHistory tableHistory = Model.getSingleton().getDb().getTableHistory();
+            RecordHistory recordHistory = getRecordHistory(tableHistory, messageId);
+
+            if (riskId < Alert.RISK_INFO || riskId > Alert.RISK_HIGH) {
+                throwInvalidAlertElementId(
+                        Constant.messages.getString(
+                                "alert.api.action.addAlert.param.riskId.illegal",
+                                PARAM_ALERT_RISK));
+            }
+            if (confidenceId < Alert.CONFIDENCE_FALSE_POSITIVE
+                    || confidenceId > Alert.CONFIDENCE_USER_CONFIRMED) {
+                throwInvalidAlertElementId(
+                        Constant.messages.getString(
+                                "alert.api.action.addAlert.param.confidenceId.illegal",
+                                PARAM_ALERT_CONFIDENCE));
+            }
+
+            HttpMessage msg = recordHistory.getHttpMessage();
```

Comment:
We need to create the `HistoryReference` as well, otherwise this will be interpreted as a new message when the alert is raised.
It could do `HttpMessage msg = getHttpMessage(messageId);` with `getHttpMessage` as:
```Java
private static HttpMessage getHttpMessage(int id) throws ApiException {
    try {
        return new HistoryReference(id, true).getHttpMessage();
    } catch (HttpMalformedHeaderException | DatabaseException e) {
        if (e.getMessage() == null) {
            throw new ApiException(ApiException.Type.DOES_NOT_EXIST, PARAM_ALERT_MESSAGE_ID);
        }
        logger.error("Failed to read the history record:", e);
        throw new ApiException(ApiException.Type.INTERNAL_ERROR, e);
    }
}
```